tez-dag/src/main/java/org/apache/tez/dag/app/rm/TaskSchedulerManager.java

```diff
@@ -910,7 +910,8 @@ public void reportError(int taskSchedulerIndex, ServicePluginError servicePlugin
       LOG.info("Error reported by scheduler {} - {}",
           Utils.getTaskSchedulerIdentifierString(taskSchedulerIndex, appContext) + ": " +
               diagnostics);
-      if (taskSchedulerDescriptors[taskSchedulerIndex].getClassName().equals(yarnSchedulerClassName)) {
+      if (taskSchedulerDescriptors[taskSchedulerIndex].getEntityName()
```

Comment:
I guess this is the actual fix

just a note, what are the values in your case:
```
taskSchedulerDescriptors[taskSchedulerIndex].getClassName()
yarnSchedulerClassName
taskSchedulerDescriptors[taskSchedulerIndex].getEntityName()
```

Reply:
Yes, this is the only actual fix.

* Before this PR

| method | return value |
| --- | ---|
| taskSchedulerDescriptors[taskSchedulerIndex].getClassName() | null |
| yarnSchedulerClassName | "org.apache.tez.dag.app.rm.YarnTaskSchedulerService" |

taskSchedulerDescriptors[taskSchedulerIndex].getClassName() is set from the variable 'taskSchedulerDescriptors' of  DAGAppMaster::serviceInit. In DAGAppMaster::parsePlugin, when we construct NamedEntityDescriptor for tez yarn plugin, the className is all null.

yarnSchedulerClassName is set from tez.am.yarn.scheduler.class, default value is "org.apache.tez.dag.app.rm.YarnTaskSchedulerService".

So for tez yarn plugin, taskSchedulerDescriptors[taskSchedulerIndex].getClassName() will never equals to yarnSchedulerClassName. Then 

* After this PR

```
taskSchedulerDescriptors[taskSchedulerIndex].getEntityName() will return "TezYarn"
```

Reply:
I see, nice catch: https://github.com/apache/tez/blob/621a83152c305ccb654f69f57fe711bc52d271be/tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java#L2676
we simply don't fill the classname, so we should not rely on it, only use it in case of createCustomTaskScheduler